### DISCUSSION TITLE
s3 Self-hosted

### DISCUSSION THREAD

**Discussion by iziwas** [OCA/odoo-community.org#100] [General]
Hi guys, I'm sorry to disturb you with this, but I face to update infrastructure in a company.
In fact, this company use OpenERP 7 and wish to update to Odoo v16. Why not ? 

1500 users could connect simultany on database and I think it's time to update their infra and put some high availability on their system.

They think to put pg_bouncer for connection pooler. In addition, of that, I think to put redis server and S3 connection for filestore and session store. Unfortunately for me, they want their services in their HQ with no external connection, so I'm thinking of MinIO. 

Could you have some retex to share with this infrastructure and some advice please?
Thanks in advance for your answers.

Best regards,
Izi

**Comment by yajo** [comment]
I'd use Ceph instead of MinIO for it's easier to scale. Other than that, those seem good choices.

**Reply by iziwas** [reply]
Thanks for your answer. I will check Ceph.

**Comment by simahawk** [comment]
Hi,

We use Minio on Azure.

You find some platform modules here
https://github.com/camptocamp/odoo-cloud-platform/

If you want to use Ceph, you should be able to add a new module to support
it.

WARNING: we've just found out that on v15 and v16 some assets gets stored
as "plain/text" mimetype. This bypasses our automatic storage rule to S3
which leads to poor performance as those assets get loaded from the DB. We
are still investigating why.

Bests

On Mon, Jul 17, 2023 at 8:38 AM Jairo Llopis ***@***.***>
wrote:

> I'd use Ceph instead of MinIO for it's easier to scale. Other than that,
> those seem good choices.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OCA/discussions/100#discussioncomment-6464346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACUYDO5T2OV6HIZ5YCD5XDXQTMVJANCNFSM6AAAAAA2L2IWXU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Simone Orsi

Full stack Python web developer, Odoo specialist, Odoo Community Board
Member, in love with open source.

**Reply by iziwas** [reply]
Oh ! Thanks for your answer, i knew odoo-cloud-platform thanks to `session_redis` module. I will check more on this repo. Thanks for the "bug" about assets. Let me know if you find something.
